backend/src/junior/junior.service.ts

```diff
@@ -66,4 +85,12 @@ export class JuniorService {
         await this.juniorRepo.save(user);
         return `${details.phoneNumber} ${content.Updated}`;
     }
+
+    private async setChallenge(phoneNumber: string): Promise<string> {
+        const valueToHash = (Math.floor(1000 + Math.random() * 9000)).toString();
```

Comment:
Does using Math.random here enable possible collisions?

Reply:
Whilst collision is possible, when checking the challenges they are matched against both a pre-assigned UUID and the hash, so potential duplicate challenges should not matter due to them being tied to a UUID.